sdk/communication/Azure.Communication.Common/tests/CommunicationIdentifierSerializerTest.cs

```diff
@@ -98,73 +99,82 @@ public void SerializeCallingApplication()
+            Assert.AreEqual(expectedIdentifier.PhoneNumber, identifier.PhoneNumber);
+            Assert.AreEqual(expectedIdentifier.Id, identifier.Id);
             Assert.AreEqual(expectedIdentifier, identifier);
         }
 
         [Test]
-        [TestCase(false)]
-        [TestCase(true)]
-        public void SerializeMicrosoftTeamsUser(bool isAnonymous)
+        [TestCase(false, null)]
+        [TestCase(true, null)]
+        [TestCase(false, "some id")]
+        [TestCase(true, "some id")]
+        public void SerializeMicrosoftTeamsUser(bool isAnonymous, string? expectedId)
         {
-            CommunicationIdentifierModel model = CommunicationIdentifierSerializer.Serialize(new MicrosoftTeamsUserIdentifier("some id", isAnonymous));
+            CommunicationIdentifierModel model = CommunicationIdentifierSerializer.Serialize(new MicrosoftTeamsUserIdentifier("user id", isAnonymous, expectedId));
 
             Assert.AreEqual(CommunicationIdentifierKind.MicrosoftTeamsUser, model.Kind);
-            Assert.AreEqual("some id", model.MicrosoftTeamsUserId);
+            Assert.AreEqual("user id", model.MicrosoftTeamsUserId);
             Assert.AreEqual(isAnonymous, model.IsAnonymous);
+            Assert.AreEqual(expectedId, model.Id);
         }
 
         [Test]
         [TestCase(false)]
         [TestCase(true)]
         public void DeserializerMicrosoftTeamsUser(bool isAnonymous)
         {
-            CommunicationIdentifier identifier = CommunicationIdentifierSerializer.Deserialize(
+            MicrosoftTeamsUserIdentifier identifier = (MicrosoftTeamsUserIdentifier)CommunicationIdentifierSerializer.Deserialize(
                 new CommunicationIdentifierModel(CommunicationIdentifierKind.MicrosoftTeamsUser)
                 {
-                    MicrosoftTeamsUserId = "some id",
+                    MicrosoftTeamsUserId = "user id",
                     IsAnonymous = isAnonymous,
+                    Id = "some id",
+                    Cloud = "gcch"
                 });
 
-            MicrosoftTeamsUserIdentifier expectedIdentifier = new MicrosoftTeamsUserIdentifier("some id", isAnonymous);
+            MicrosoftTeamsUserIdentifier expectedIdentifier = new MicrosoftTeamsUserIdentifier("user id", isAnonymous, "some id", CommunicationCloudEnvironment.Gcch);
 
-            Assert.True(identifier is MicrosoftTeamsUserIdentifier);
-            Assert.AreEqual(expectedIdentifier.UserId, ((MicrosoftTeamsUserIdentifier)identifier).UserId);
-            Assert.AreEqual(expectedIdentifier.IsAnonymous, ((MicrosoftTeamsUserIdentifier)identifier).IsAnonymous);
+            Assert.AreEqual(expectedIdentifier.UserId, identifier.UserId);
+            Assert.AreEqual(expectedIdentifier.IsAnonymous, identifier.IsAnonymous);
+            Assert.AreEqual(expectedIdentifier.Cloud, identifier.Cloud);
+            Assert.AreEqual(expectedIdentifier.Id, identifier.Id);
             Assert.AreEqual(expectedIdentifier, identifier);
         }
     }
```

Comment:
Do we have a case that tests that we throw when `Cloud` is missing during deserialization?

Reply:
yes